### PR TITLE
Cloudrun v2 job

### PR DESCRIPTION
**Pull request by shuyama1**
This should resolve https://github.com/hashicorp/terraform-provider-google/issues/11743
Internal ticket: b/240175294
API doc can be found in the internal ticket

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_cloud_run_v2_job`
```

### CODE REVIEW

File: mmv1/templates/terraform/examples/cloudrunv2_job_sql.tf.erb [hunk 1]
```diff
@@ -0,0 +1,71 @@
+resource "google_cloud_run_v2_job" "<%= ctx[:primary_resource_id] %>" {
+  name     = "<%= ctx[:vars]['cloud_run_job_name'] %>"
+  location = "us-central1"
+  launch_stage = "BETA"
+  
+  template {
+    template{
```
Comment (jeremad) [inline mmv1/templates/terraform/examples/cloudrunv2_job_sql.tf.erb:7]:
```suggestion
    spec {
```
?

Comment (shuyama1) [reply mmv1/templates/terraform/examples/cloudrunv2_job_sql.tf.erb:7]: This job resource is implemented using v2 API, which I believe `spec` got flattened and `template { template {}}` should be the correct format.

**Comment by bogaertg** [comment]
up ?

**Review by roaks3 — APPROVED** [review]
Looks great! A vast majority of this matches the cloudrun service, and from what I saw, everything looked consistent.

Only nit is that I didn't see some fields in the update test (for example, `container.command` and `livenessProbe.tcpSocket`), but I don't know if we generally aim for full coverage there.

Adding Stephen for follow-up

File: mmv1/products/cloudrunv2/api.yaml [hunk 2]
```diff
@@ -0,0 +1,660 @@
...
+                          - !ruby/object:Api::Type::String
+                            name: "name"
+                            required: true
+                            description: |-
+                              Name of the environment variable. Must be a C_IDENTIFIER, and mnay not exceed 32768 characters.
+                          - !ruby/object:Api::Type::String
+                            name: "value"
+                            description: |-
+                              Variable references $(VAR_NAME) are expanded using the previous defined environment variables in the container and any route environment variables. If a variable cannot be resolved, the reference in the input string will be unchanged. The $(VAR_NAME) syntax can be escaped with a double $$, ie: $$(VAR_NAME). Escaped references will never be expanded, regardless of whether the variable exists or not. Defaults to "", and the maximum length is 32768 bytes
+                            # exactly_one_of:
```
Comment (melinath) [inline mmv1/products/cloudrunv2/api.yaml:194]: Copying from the cloud run service PR:

> `ConflictsWith`, `ExactlyOneOf` and `AtLeastOneOf` within lists are currently not supported at the SDK level.

File: mmv1/products/cloudrunv2/api.yaml [hunk 3]
```diff
@@ -0,0 +1,660 @@
...
+                                required: true
+                                description: |-
+                                  The relative path of the secret in the container.
+                              - !ruby/object:Api::Type::String
+                                name: "version"
+                                description: |-
+                                  The Cloud Secret Manager secret version. Can be 'latest' for the latest value or an integer for a specific version
+                              - !ruby/object:Api::Type::Integer
+                                name: "mode"
+                                required: true
```
Comment (melinath) [inline mmv1/products/cloudrunv2/api.yaml:456]: it seems from the description like this is not required

```suggestion
```

**Review by melinath — APPROVED** [review]
LGTM - one question below.

Adding tests for the fields Ryan mentioned above does seem like it would be nice to do.

File: mmv1/products/cloudrunv2/api.yaml (hunk 3 above)
Comment (shuyama1) [reply mmv1/products/cloudrunv2/api.yaml:456]: yeah, it does look like it's not required. I'll test it and remove it if the requirement is not necessary.

**Comment by shuyama1** [comment]
Thanks @roaks3 and @melinath! I'll add more test coverage.

File: mmv1/products/cloudrunv2/api.yaml (hunk 3 above)
Comment (shuyama1) [reply mmv1/products/cloudrunv2/api.yaml:456]: Tested it and can confirm that `mode` is required.

**Review by melinath — APPROVED** [review]

**Comment by shuyama1** [comment]
cl/492014758 which disallows job.name in v2 CreateJobRequest is submitted and will be rolled out next week. Therefore, I marked the field `url_param_only`. (name is long form output only in API but in Terraform we implement it as short-form name)

**Review by melinath — APPROVED** [review]

**Pull request merged**